submissions/mxmgny/a-tiny-js-world/index.js

```diff
@@ -0,0 +1,43 @@
+let cat = {
+    appearance: "cat",
+    name: "Garfield",
+    legs: 4,
+    hands: 0,
+    gender: "male",
+    shout: "I love lazagna!",
+    friends:['Selina', 'Jeremy']
+};
+let cat_woman = {
+    appearance:"cat-woman",
+    name:"Selina",
+    legs:2,
+    hands:2,
+    gender: "female",
+    shout:cat['shout'],
+    friends: [cat['name'],'Sansa']
+};
+let dog = {
+    appearance:"dog",
+    name: "Jerry",
+    gender: "male",
+    legs: 4,
+    hands: 0,
+    shout: "Woof!",
+    friends: ["Sansa","Snoop Dog", "Balto", "Goofy"]
+};
+let human = {
+    appearance:"human",
+    name:"Sansa",
+    gender:"female",
+    legs:2,
+    hands:2,
+    shout:"Winter is coming",
+    friends:[dog['name'], "Robb", "Arya", "Bran", "Rickon"]
+};
+
+let arr = [cat, dog, human,cat_woman];
+
+arr.forEach(obj => {
+    let friends = obj['friends'].join(', ');
+    print(`Hi, my name is <strong>${obj['name']}</strong>. I am <em>${obj['gender']} ${obj['appearance']}</em>.I have ${obj['legs']} legs and ${obj['hands']} hands. My friends are ${friends}. And you know what? ${obj['shout']}`);
+})
```

Comment:
Good to see that you're already familiar with ES6 Feature - String Interpolation!
Let's improve your pattern so it could be more readable and maintainable.
First of all - I don't think that `let friends = obj['friends'].join(', ');` is needed action. You can rid of `friends` variable declaration and just put `${obj.friends.join(', ')}` in to your pattern as it's still short and readable thing.

The second one is about organizing.
Writing the code in one very long line - bad practice. Fortunately, it is never a necessary method.
What if instead of a single string we put an array including that string's parts in a `print` function and `.join` them?

**Little hint:**
[
\`${obj.name} some text,\`
\`${obj.gender} some text,\`
].join('')

Reply:
Hi, thank you for your comments. It helps me to understand my mistakes and improve my skill. I think I changed everything you asked for. I'm sorry for taking it too long.